### PR TITLE
Additions to support autoconfiguration of connected displays

### PR DESCRIPTION
**Pull request by cormacc**
Thanks for this gem. Got annoyed with updating my laptops wm config file with different xrandr command lines to cope with changes in drivers and connected displays so wrote a little autoconfiguration tool around it. May or may not be of interest, but have done the following:

Added a new Xrandr::Configurator class for display autoconfiguration. It
currently features a single fairly opinionated heuristic that suits my
own setup (laptop that may have one or more external displays
connected), but underlying logic adaptable to others.

Added a basic Xrandr::Resolution class that converts the resolution
string into numeric X and Y components and implements the comparable
interface, allowing displays to be sorted on the basis of total pixel
count

Made the following (I think compatible) modifications to existing
classes...

Xrandr::Control
Added convenience methods 'connected_outputs' and 'disconnected_outputs'

Xrandr::Mode
Updated initializer to construct an instance of the new Resolution
class. The Xrandr::Resolution.to_s method degrades to the original
implementation, so should be compatible with existing API.

I wrap the new configurator class using Thor for command line use, but reckoned there was probably no point in adding the dependency to your gem? Can amend the PR if you reckon it's worth pushing in.

### CODE REVIEW

**Comment by F-3r** [comment]
Hey! Glad you find it useful :)  
I'll take a deeper look to the changes in this PR in these days.

Doing a fast review, this library is intended to be as low level and as general as possible (a tool to create other tools). Taking that into account, I believe the `Configurator` class would be out of the scope of these gem.  On the other hand, `Resolution` looks like we can include it. 

I'll probably request a few changes if that's not a problem for you :)

Thank you very much for the contribution!

**Comment by cormacc** [comment]
Reckoned the Configurator might be out of scope alright - kept it in a separate source file so it would be easy to strip out.

Changes not a problem - let me know what you want done.

File: lib/xrandr.rb [hunk 1]
```diff
@@ -36,6 +36,16 @@ def apply!
...
       initialize
     end
+
+    def connected_outputs
+      outputs.select { |output| output.connected }
+    end
+
+    def disconnected_outputs
+      outputs.reject { |output| output.connected }
+    end
```
Comment (F-3r) [inline lib/xrandr.rb:46]: :+1:  we have this logic repeated many times in our application

File: lib/xrandr.rb [hunk 2]
```diff
@@ -36,6 +36,16 @@ def apply!
...
+
+
```
Comment (F-3r) [inline lib/xrandr.rb]: :policeman:  THE CODE STYLE COP says:  _Would you please remove this extra white line?_

File: lib/xrandr.rb [hunk 3]
```diff
@@ -154,10 +164,37 @@ class Mode
...
+  ##
+  # This class converts a resolution string of the form "1234x5678" (arbitrary
+  # digit count) into a form with discrete numeric x and y components.
+  # It implements the comparable interface on the basis of total pixel count.
+  class Resolution
+    include Comparable
+    MATCHER = /(?<x>\d+)x(?<y>\d+)/
+    attr_accessor :x
+    attr_accessor :y
```
Comment (F-3r) [inline lib/xrandr.rb]: I would rather have this 2 as `attr_reader` (don't see much value in letting this instances be modified from outside)

Also, I believe the names we use for resolution attributes are :width and :height 

Finally, the CODE STYLE COP :policeman:  says he prefers attributes in a single line: `attr_reader :width, :height`

**Review by F-3r — APPROVED** [review]
@cormacc Thank you very much for this PR (and sorry for the delay in my review).

Small comments aside, It looks good :) 
Feel free to add them to your branch and we'll merge it right away :)

**Comment by cormacc** [comment]
One minor addition -- replaced the repeated width*height in Resolution<=> with a call to a new 'abs' function.

**Pull request merged**